### DISCUSSION TITLE
Linear taper with cladding

### DISCUSSION THREAD

**Discussion by JinchW** [gdsfactory/gdsfactory#2946] [Q&A] [answered]
Hi, I'm not sure if it's a problem within the code. When I try to draw a taper with cladding, the cladding also appears on both ends of the port, but that's not what I want.I checked the cross_section example, and there shouldn't be an offset on both ends unless you select bbox. Following is my code (gdsfactory 8.4.0 & python 3.10):

```
from functools import partial
import gdsfactory as gf

new_strip = gf.partial(
gf.cross_section.cross_section,
width=0.5,
cladding_layers = ['WGCLAD'],
cladding_offsets = [5],
)

taper = gf.components.taper(
length = 10,
width1 = 0.5,
width2 =1,
cross_section = new_strip
)

taper.write_gds("demo.gds")
taper.show()
```
I think the cladding in the red box is not necessary, which will cause DRC errors.
![github](https://github.com/gdsfactory/gdsfactory/assets/107834407/6ca475eb-4932-4bac-bb55-690f738b71fe)

**Comment by sebastian-goeldi** [comment]
@joamatab ?

**Comment by joamatab** [comment]
yes, this is a bug, thank you @JinchW  for reporting this!

See fix [here](https://github.com/gdsfactory/gdsfactory/pull/2970)

**Reply by JinchW** [reply, marked as answer]
Tkx, it really works!
